index.bs

```diff
@@ -0,0 +1,52 @@
+<pre class='metadata'>
+Title: WebCodecs
+Repository: wicg/web-codecs
+Status: CG-DRAFT
+ED: https://wicg.github.io/web-codecs/
+Shortname: web-codecs
+Level: 1
+Group: wicg
+Editor: Chris Cunningham, w3cid 114832, Google Inc. https://google.com/
+Abstract: WebCodecs is a flexible web API for encoding and decoding audio and video.
+!Participate: <a href="https://github.com/wicg/web-codecs">Git Repository.</a>
+!Participate: <a href="https://github.com/wicg/web-codecs/issues/new">File an issue.</a>
+!Version History: <a href="https://github.com/wicg/web-codecs/commits">https://github.com/wicg/web-codecs/commits</a>
+</pre>
+
+<section class='non-normative'>
+  <h2 id='introduction'>Introduction</h2>
+  <em>This section is non-normative</em>
+  <p>
+    The introduction has not yet been written.
+  </p>
+  <p class='note'>
+    NOTE: The interfaces that follow are a work in progress.
+  </p>
+</section>
+
+<section>
+  <h2 id='video-decoder-interface'>VideoDecoder Interface</h2>
+  <pre class='idl'>
+    [Exposed=(Window)]
+    interface VideoDecoder {
+        constructor();
+        Promise&lt;void&gt; Initialize(VideoDecoderInitParameters params);
+        Promise&lt;void&gt; Flush();
+        void Close();
+
+        readonly attribute WritableStream writable;  // of EncodedVideoFrame
+        readonly attribute ReadableStream readable;  // of VideoFrame
+    };
+  </pre>
+  <pre class='idl'>
+    [Exposed=(Window)]
+    dictionary VideoDecoderInitParameters {
+      required DOMString codec;
+      required DOMString profile;
+
+      // These are the "coded size", not the "visible size"
+      required unsigned long width;
+      required unsigned long height;
+    };
+  </pre>
```

Comment:
For VideoEncoder, I think you can add a subset I've got in my CL for now:

interface VideoEncoder {
    Promise<void> Initialize(VideoEncoderInitParameters params);
    void Close();

    readonly attribute WritableStream writable;  // of VideoFrame
    readonly attribute ReadableStream readable;  // of EncodedVideoFrame
};

enum VideoEncoderHardwareMode {
  "hardware",
  "software"
};

dictionary VideoEncoderInitParameters {
  VideoEncoderHardwareMode hardware_mode;
  DOMString codec;
  DOMString profile;

  unsigned long visible_width;
  unsigned long visible_height;

  unsigned long bitsPerSecond;
  double framesPerSecond;

  double maxFramesBetweenKeyFrames;
};

Reply:
Done!